src/vm/syncblk.cpp

```diff
@@ -2564,6 +2564,42 @@ double ComputeElapsedTimeInNanosecond(LARGE_INTEGER startTicks, LARGE_INTEGER en
     return (elapsedTicks * NsPerSecond) / freq.QuadPart;
 }
 
+DWORD AwareLock::WaitForSemEvent(INT32 timeOut)
+{
+    // We might be interrupted during the wait (Thread.Interrupt), so we need an
+    // exception handler round the call.
+    struct Param
+    {
+        AwareLock *pThis;
+        INT32 timeOut;
+        DWORD ret;
+    } param;
+    param.pThis = this;
+    param.timeOut = timeOut;
+
+    EE_TRY_FOR_FINALLY(Param *, pParam, &param)
```

Comment:
It would be better to rewrite it using object with a destructor that does `m_lockState.InterlockedUnregisterWaiter();`